### PR TITLE
fix regExp for spaces separators

### PR DESCRIPTION
**Pull request by DaniilMoiseevPenza**
## Proposed changes
Some space separator don`t deletes from Strings in texts assertions

## Checklist
- [x] Checkstyle and unit tests are passed locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by DaniilMoiseevPenza** [comment]
@BorisOsipov @asolntsev 
https://t.me/selenide_ru/69754

**Review by asolntsev — CHANGES_REQUESTED** [review]
@DaniilMoiseevPenza Can you please check why many tests are failing?

And I didn't understand why you added `a-z0-9` to the regexp.
Doesn't it mean that all latin characters are spaces?

For example,
```
  Html.text.reduceSpaces("Bruce   \n\t   Willis")
```

returns `"B W"` instead of `"Bruce Willis"`. :(

**Comment by DaniilMoiseevPenza** [comment]
@asolntsev 
Yep)
Now write new more possible regular)

**Comment by asolntsev** [comment]
> Now write new more possible regular)

Sorry, I don't understand your answer :(

**Comment by DaniilMoiseevPenza** [comment]
В процессе написания более правильной регулярки))

**Comment by asolntsev** [comment]
> В процессе написания более правильной регулярки))

Hint:

image

**Comment by DaniilMoiseevPenza** [comment]
@asolntsev @BorisOsipov 
fix)

**Comment by BorisOsipov** [comment]
@DaniilMoiseevPenza 

1) I've updated the test to actually inject Unicode symbol to the test string(update your local branch to get it). Current test injects unicode **number** instead of actual **symbol**. So without good tests there is no good implementation. The regex matches unicode numbera instead of symbols and all tests failed. Write good tests first!
2) `com.codeborne.selenide.conditions.TextTest#apply_for_textInput_ignoresWhitespaces` is not the best place to add new test.  There is `com.codeborne.selenide.impl.TextsTest` when we actually test the  `reduceSpaces()` method against different space symbols.
3) You don't really need that `SpaceSeparator` enum. Using `@CsvSource` with a test file is much easier. 
4) Probably regex can be wider and less complicated. `\p{Z}`  or `\p{Zs}` are a shortcuts for all space-like Unicode symbols. Consider using them.

**Comment by DaniilMoiseevPenza** [comment]
> @DaniilMoiseevPenza
> 
> 1. I've updated the test to actually inject Unicode symbol to the test string(update your local branch to get it). Current test injects unicode **number** instead of actual **symbol**. So without good tests there is no good implementation. The regex matches unicode numbera instead of symbols and all tests failed. Write good tests first!
> 2. `com.codeborne.selenide.conditions.TextTest#apply_for_textInput_ignoresWhitespaces` is not the best place to add new test.  There is `com.codeborne.selenide.impl.TextsTest` when we actually test the  `reduceSpaces()` method against different space symbols.
> 3. You don't really need that `SpaceSeparator` enum. Using `@CsvSource` with a test file is much easier.
> 4. Probably regex can be wider and less complicated. `\p{Z}`  or `\p{Zs}` are a shortcuts for all space-like Unicode symbols. Consider using them.

Thank you, i will fix)

**Comment by DaniilMoiseevPenza** [comment]
@BorisOsipov @asolntsev 
fix)

**Review by asolntsev — APPROVED** [review]

**Comment by asolntsev** [comment]
@DaniilMoiseevPenza Thank you for the contribution. It has been released in Selenide 7.6.0.

Also, I added PR https://github.com/selenide/selenide/pull/2884 for additional features with spaces.
